modules/apps/portal-workflow/portal-workflow-kaleo-test/src/testIntegration/java/com/liferay/portal/workflow/kaleo/internal/upgrade/v4_2_1/test/WorkflowContextUpgradeProcessTest.java

```diff
@@ -0,0 +1,161 @@
+import com.liferay.portal.kernel.test.util.ServiceContextTestUtil;
+import com.liferay.portal.kernel.test.util.TestPropsValues;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.kernel.util.HashMapBuilder;
+import com.liferay.portal.kernel.workflow.WorkflowConstants;
+import com.liferay.portal.test.log.LogCapture;
+import com.liferay.portal.test.log.LoggerTestUtil;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.test.rule.PermissionCheckerMethodTestRule;
+import com.liferay.portal.upgrade.registry.UpgradeStepRegistrator;
+import com.liferay.portal.upgrade.test.util.UpgradeTestUtil;
+import com.liferay.portal.workflow.kaleo.model.KaleoInstance;
+import com.liferay.portal.workflow.kaleo.service.KaleoInstanceLocalService;
+
+import java.io.Serializable;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Jhosseph Gonzalez
+ */
+@RunWith(Arquillian.class)
+public class WorkflowContextUpgradeProcessTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerMethodTestRule.INSTANCE);
+
+	@Before
+	public void setUp() throws Exception {
+		WorkflowDefinitionLinkLocalServiceUtil.addWorkflowDefinitionLink(
+			TestPropsValues.getUserId(), TestPropsValues.getCompanyId(),
+			TestPropsValues.getGroupId(), "com.liferay.blogs.model.BlogsEntry",
+			0, 0, "Single Approver", 1);
+
+		Group group = GroupTestUtil.addGroup();
+
+		_blogsEntryLocalService.addEntry(
+			TestPropsValues.getUserId(), RandomTestUtil.randomString(),
+			RandomTestUtil.randomString(),
+			ServiceContextTestUtil.getServiceContext(group.getGroupId()));
+
+		ServiceContext serviceContext = new ServiceContext();
+
+		serviceContext.setCompanyId(TestPropsValues.getCompanyId());
+		serviceContext.setUserId(TestPropsValues.getUserId());
+
+		KaleoInstance kaleoInstance =
+			_kaleoInstanceLocalService.addKaleoInstance(
+				1, 1, RandomTestUtil.randomString(), 1,
+				HashMapBuilder.<String, Serializable>put(
+					WorkflowConstants.CONTEXT_ENTRY_CLASS_NAME,
+					"com.liferay.blogs.model.BlogsEntry"
+				).put(
+					WorkflowConstants.CONTEXT_SERVICE_CONTEXT,
+					(Serializable)serviceContext
+				).build(),
+				serviceContext);
+
+		JSONObject workflowContextJSONObject = JSONFactoryUtil.createJSONObject(
+			kaleoInstance.getWorkflowContext());
+
+		JSONObject mapJSONObject = workflowContextJSONObject.getJSONObject(
+			"map");
+
+		mapJSONObject.remove("javaClass");
+
+		JSONObject serviceContextJSONObject = mapJSONObject.getJSONObject(
+			"serviceContext");
+
+		serviceContextJSONObject.put(
+			"javaClass",
+			"com.liferay.headless.common.spi.service.context." +
+				"ServiceContextUtil$1");
```

Comment:
@JhossephG the wrong `json` looks like this:

```json
"map": {
    "serviceContext": {
        "javaclass": "com.liferay.headless.common.spi.service.context.ServiceContextUtil$1",
        "parameter1" "",
        "parameter2" ""
    }
}

```

and the correct `json` looks like this:

```json
"map": {
    "serviceContext": {
        "javaclass": "com.liferay.portal.kernel.service.ServiceContext",
        "serializable": {
            "parameter1" "",
            "parameter2" ""
        }
    }
}
```

1. So why are you removing `javaClass` from `map` if it only contains `serviceContext`? 
2. And why are you not manipulating the `serializable`?